### PR TITLE
Detect Intel 5200U broadwell as haswell

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by wbhart** [comment]
Thanks for this, but it isn't in line with the way MPIR is structured. Broadwell is a revision of Haswell and introduces new instructions. This means that there should be a directory broadwell inside the haswell directory. cpuid should then return broadwell as the model. Configure.ac should then select first the broadwell directory first (where there will at least be a gmp-mparam.h file with broadwell specific tuning values), followed by the haswell directory, etc. This will allow us to add broadwell specific assembly at a later point.

**Comment by isuruf** [comment]
I don't know how that works with `haswell/avx`. I can provide you with a `gmp-mparam.h` if that helps.

**Comment by wbhart** [comment]
I think some Haswells actually support these new instructions. If that
isn't the case, we should perhaps do a directory rename. But I vaguely
recall the guy who set this up saying that much to his surprise, some
Haswells did in fact support the new instructions.

**Comment by wbhart** [comment]
Sorry, I was confused here. Intel ADX was first supported by Broadwell. I
was thinking of Skylake, where MPIR now uses Intel ADX. Haswell does not.

Instead, the haswell/avx directory is for Haswells that support AVX2, etc.
Not all Haswells do. Only the core i3, i5 and i7 Haswells do this. The
cpuid detects this and ensures that only Haswells that support AVX2 etc.,
actually use the haswell/avx directory. But this is independent of the
Haswell/Broadwell distinction.

Basically, broadwell would first take code from haswell/broadwell, then
haswell/avx, then the other directories in order that haswell uses (see how
haswell is configured in configure.ac -- just search for haswell).

**Comment by wbhart** [comment]
As a minimum, attaching your tuning values to a ticket so that someone else
can add support for your system, that would be helpful. Of course it is
hard for us to support systems we don't have ourselves.
